service/currencies.js

```diff
@@ -0,0 +1,60 @@
+var _ = require('lodash');
+
+module.exports = {
+    getCandles: function(data, totime, fromTime, interval, currency, callback) {
+        var result = []
+        var i = 0
+        //using loop till we filter all the arrray chunks
+        do {
+            // getting array chunks for every time interval
+            var arrayChunks = _.filter(data, function(result) {
+                return ((new Date(totime) >= new Date(result.date)) && (new Date(fromTime) <= new Date(result.date)));
+            })
+            //if filter result is not empty
+            if (arrayChunks.length > 0) {
+                //storing the prices of first minute of a time interval, to use for further calculations
+                var prices = arrayChunks[0].price
+                //calculating volume by getting difference between last iteration volume and current iteration volume
+                //and storing the high,low and volume of last data in array
+                for (var a in prices) {
```

Comment:
i think its more easier to understand if you do for(j=0;k<prices.lenght;  type of for loop here

because index are 0,1,2, etc

Reply:
ok sir